Do a manual (non-cli) upload

amvanbaren opened the pull request:
Added publish button to header
Added extensions tab in user settings
Added ability to publish extension
Added ability to create namespace
Make namespaces visible for contributors

Testing steps:
A publish button has been added to the navigation menu.
Click the publish button, the rest of the UX should speak for itself.

amvanbaren commented:
@jeanp413 @filiptronicek Can you try out this new feature?

filiptronicek reviewed (changes requested):
Huge thanks for the effort on this feature, @amvanbaren 🙏.

Here's the feedback I have so far:
- I really like the extension upload dialog, but it would be nice to make it clear that the user can only upload one extension at a time and if the user tries to upload more there should be a warning / alert.
- Maybe we can address this in a future PR, but it would be great to better handle non-existent namespaces in the UI (e.g. let the user create the namespace or create it automagically)
![image](https://user-images.githubusercontent.com/29888641/164787895-75ca1e74-ad66-4137-91d8-e89b5d41d5c6.png)

- there is an issue with text contrast on the `Create namespace` modal when it is focused 

| No focus | Focus |
|-----|------------|
| ![image](https://user-images.githubusercontent.com/29888641/164788267-0ad28ae6-a3c1-4c90-afb0-b7bedf4c7682.png) | ![image](https://user-images.githubusercontent.com/29888641/164788546-3a68aa58-8323-4959-8165-4e104d279b85.png) | 

Other than that, the UX is very intuitive and it was very easy for me to make the publishing happen.

amvanbaren commented:
@filiptronicek 
>  it would be nice to make it clear that the user can only upload one extension at a time and if the user tries to upload more there should be a warning / alert.

Dropzone is configured to only allow one file. I've changed the dropzone text to make that clearer. An existing file is overwritten when a user picks another file. Instead of a warning (which can get annoying) I've added an undo action in case the user didn't mean to change the extension package to be uploaded.

> it would be great to better handle non-existent namespaces in the UI (e.g. let the user create the namespace or create it automagically.

I've added error handling that tries to create the namespace automagically and if successful publish the extension again.

> there is an issue with text contrast on the Create namespace modal when it is focused

This was a site-wide issue where text fields didn't look right when focused. The same happened when writing a review, generating an access token or adding a namespace member. It's fixed now.

filiptronicek commented on webui/src/pages/user/upload-extension-dialog.tsx:
```diff
@@ -0,0 +1,273 @@
+        const successClass = this.context.pageSettings.themeType === 'dark' ? classes.successDark : classes.successLight;
+
+        return <React.Fragment>
+            <Button variant='outlined' onClick={this.handleOpenDialog}>Upload extension</Button>
```
Maybe we should rethink the wording on these dialogs and buttons. I think if we tell the users that with this button you "upload" an extension, they would think it would not publish it, just upload it as a "draft" or similar. 

I, personally would reword this as "Publish Extension", as well as change the "Upload" button to "Publish" in the modal upload dialog.

filiptronicek replied:
And maybe also add a little note in the modal, saying something like "After you click 'Publish', this extension will be available on the Marketplace"

amvanbaren replied:
I've renamed everything from upload to publish.
I added a title to the publish button in the dialog.

filiptronicek commented:
Otherwise loving the process and publishing flow, really a delightful experience! 🧡

amvanbaren commented:
> I noticed that the navbar has the Publish item when you are not in the user dashboard. This **may** be confusing to some users because otherwise the navbar is the same across the whole website and now it disappears here. I think it is nice having it, I would keep it and have it everywhere for sake of inconsistency, but I'd love to know what you think, @amvanbaren 🙂.

Ok, the publish button is now also available in `user-settings`.

filiptronicek reviewed (approved):
Thanks for addressing my comments, changes look good to me. :shipit: ✅
